akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/EventsourcedRequestingRecoveryPermit.scala

```diff
@@ -0,0 +1,96 @@
+/**
+ * Copyright (C) 2016-2018 Lightbend Inc. <http://www.lightbend.com/>
+ */
+package akka.persistence.typed.internal
+
+import akka.actor.typed.Behavior
+import akka.actor.typed.scaladsl.Behaviors.MutableBehavior
+import akka.actor.typed.scaladsl.{ ActorContext, Behaviors, StashBuffer, TimerScheduler }
+import akka.annotation.InternalApi
+import akka.event.Logging
+import akka.persistence._
+import akka.persistence.typed.internal.EventsourcedBehavior.WriterIdentity
+
+/**
+ * INTERNAL API
+ *
+ * First (of four) behaviour of an PersistentBehaviour.
+ *
+ * Requests a permit to start recovering this actor; this is tone to avoid
+ * hammering the journal with too many concurrently recovering actors.
+ */
+@InternalApi
+private[akka] final class EventsourcedRequestingRecoveryPermit[Command, Event, State](
+  val persistenceId:    String,
```

Comment:
It seems these parameters are the same for all the stages. Either put them all into a `Setup` class or we could also go for a cake pattern where the cake provides these values and all the state behaviors are inside the cake?

Reply:
Such setup I introduced for other things already, I can include the persistenceId as well I guess

Reply:
> we could also go for a cake pattern where the cake provides these values and all the state behaviors are inside the cake?

Spent half a day today migrating away from the cake though... was not that nice.